wordpress-mobile/WordPress-Android
# Refactors for notifications

Part of pc8HXX-1ua-p2

This PR contains:
1. Extract NoteViewHolder from NoteAdapter
2. Use ViewBinding in NoteViewHolder
3. Remove a duplicated API request from NotificationsListFragment
4. Fix IndexOutOfBoundsException in NoteAdapter
5. Refactors

-----

## To Test:
1. Sign in the JP app and switch to Notifications tab
2. Click some notification items
3. The behavior should be the same as before.
4. Switch to ALL/UNREAD/COMMENTS...etc
5. Repeat 2 & 3
6. The behavior should be the same as before, thank you.

P.S. The original implementation caused an IndexOutOfBoundsException after switching to ALL/UNREAD/COMMENTS tabs multiple times and then clicking on the notification item.



-----

## Regression Notes

1. Potential unintended areas of impact

    - notifications tab

8. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

9. What automated tests I added (or what prevented me from doing so)

    - none

-----

## PR Submission Checklist:


-----

## Testing Checklist:
